### DISCUSSION TITLE
Existing Cookies not passed in request

### DISCUSSION THREAD

**Discussion by benm5678** [cefsharp/CefSharp#4511] [General]
We are using this package for a long time to automate getting info from some Url that relies on existing cookies from user's Chrome to work.  It seems suddenly it no longer passes the cookies -- we have a RequestHandler hooked up to ChromiumWebBrowser, and when I inspect the Request object headers while in OnResourceResponse, I don't see Cookies being passed.

I init Cef with the following... I also tried setting UserDataPath & CachePath & PersistSessionCookies - but it doesn't help.  We were using version 96.* and I tried upgrading to latest and still same issue.  I suspect something in latest Chrome updates broke it, as it was working for many people in our organization for several years.  Any ideas?

```
var settings = new CefSettings { IgnoreCertificateErrors = true };
settings.CefCommandLineArgs.Add("auth-server-whitelist", "*");
settings.CefCommandLineArgs.Add("auth-delegate-whitelist", "*");
CefSharp.Cef.Initialize(settings)
```

**Comment by BuslikDrev** [comment]
Домен как назвали вы? Доменную зону нужно указать ("*.com" or "*.by" or "*.ua" or others). CefSharp v109.1.110.0 работают cookie.
После изменения домена советую удалить кэш в CachePath.
```
CefSharp.WinForms.CefSettings settings = new CefSharp.WinForms.CefSettings() { IgnoreCertificateErrors = true };
settings.CefCommandLineArgs.Add("auth-server-whitelist", "*");
settings.CefCommandLineArgs.Add("auth-delegate-whitelist", "*");

				settings.RootCachePath = "D:\\myapp\\cache";
				settings.CachePath = "D:\\myapp\\cache";
				settings.UserDataPath = "D:\\myapp\\userdata";
				settings.RegisterScheme(new CefSharp.CefCustomScheme {
					SchemeName = "https",
					DomainName = "mydomain.loh",
					SchemeHandlerFactory = new CefSharp.SchemeHandler.FolderSchemeHandlerFactory (
						rootFolder: "D:\\myapp\\",
						hostName: "mydomain.loh",
						defaultPage: "index.html"
					)
				});

CefSharp.Cef.Initialize(settings);

CefSharp.WinForms.ChromiumWebBrowser browser = new CefSharp.WinForms.ChromiumWebBrowser('https://mydomain.loh/index.html');
```

**Reply by benm5678** [reply]
To clarify, I need it to use cookies from user's default Chrome, not from some cache path specific to the app.  So if a user opened Chrome in their machine and got some cookies from IdP, those cookies should be available when someone uses the tool (i.e. when it accesses the IdP it should pass the cookies).  It used to work...not sure what changed.

**Reply by BuslikDrev** [reply]
Я думаю это может быть сделано в целях безопасности, чтобы не смогли с помощью Chromium воровать cookie Chrome (странно, что вообще такое можно было). Можете спросить у создателей Chrome.

**Reply by amaitland** [reply]
> I need it to use cookies from user's default Chrome

That's never officially been supported, if it worked then it was pure luck. From a security point of view that would be concerning, 3rd party applications could steal `Chrome cookies`. I expect you'll have errors in your log file saying it's unable to access the profile/cache.

Additionally Chromium only allows a single process to access a cache(profile). Even if this was supported, if Chrome was open then you'd be unable to have your process (application) access the cache.

**Reply by benm5678** [reply]
Yes, you are right @amaitland ... it seems it relies on our windows credentials instead.  So in my regular Chrome even if I delete all cookies, I can still get to the auth url.  I saw a post saying in Chrome 110 auth-server-whitelist was finally dropped... so I tried updating to auth-server-allowlist, but it didn't seem to help.  Is anyone having luck with the new switch?

**Reply by benm5678** [reply]
I finally found by adding below fixed the issue.  Not sure what happened there, but more likely to some corporate network change on our end.  Sorry to bug here, appreciate the nice support!
`settings.CefCommandLineArgs.Add("proxy-auto-detect", "1");`
